alacritty/src/display.rs

```diff
@@ -241,11 +242,11 @@ impl Display {
         info!("Cell Size: {} x {}", cell_width, cell_height);
         info!("Padding: {} x {}", padding_x, padding_y);
 
-        // Create new size with at least one column and row.
+        // Create new size with at least 2 columns and 1 row.
         let size_info = SizeInfo {
             dpr,
-            width: (viewport_size.width as f32).max(cell_width + 2. * padding_x),
-            height: (viewport_size.height as f32).max(cell_height + 2. * padding_y),
+            width: (viewport_size.width as f32).max(cell_width * MIN_COLS as f32 + 2. * padding_x),
+            height: (viewport_size.height as f32).max(cell_height * MIN_LINES as f32 + 2. * padding_y),
```

Comment:
We must not ever change the size_info passed to the renderer/input. If this is limited, nothing will work anymore with small sizes.

All we want to do is not shrink the grid.

So the comment of @kchibisov is correct, but using `set_min_inner_size` isn't really the problem, since we already use `with_inner_size` in window anyways.

Setting a minimum inner size is basically completely pointless, since everyone can ignore that anyways. And Alacritty still functions okay without that.

Reply:
> So the comment of @kchibisov is correct, but using set_min_inner_size isn't really the problem, since we already use with_inner_size in window anyways.

>Setting a minimum inner size is basically completely pointless, since everyone can ignore that anyways. And Alacritty still functions okay without that.

It’s not incorrect, it’s a hint to not shrink you any lower than certain size. So it could be useful and is kind of respected/works on mostly all systems. So it’s nice to set it, I guess, but not required for that PR.

Reply:
Yeah, it's a nice to have, regardless of the issue of this PR. It's just useful for communicating the minimum size to our users.

Reply:
Guess I should keep them then?

Reply:
>Guess I should keep them then?

it was wrt `set_min_inner_size`. Your updates to display/renderer should be removed. You may replace them with `set_min_inner_size` though, since  those are fine to use.